docs/scoping-and-declarations.rst

```diff
@@ -73,6 +73,44 @@ You cannot directly declare tuple types. However, in certain cases you can use l
         a, b = self.foo()
 
 
+Storage Layout
+==============
+
+Storage variables are located within a smart contract at specific storage slots. The compiler allocates the first variable to be stored within `slot 0` and subsequent variables are stored in order.
+
+There are cases where it is necessary to override this pattern and to allocate storage variables in custom slots. This behaviour is often required for upgradeable contracts, to ensure that both contracts (the old contract, and the new contract) store the same variable within the same slot.
+
+This can be performed when compiling  via ``vyper`` by including the ``--storage-layout-file`` flag.
+
+For example, consider upgrading the following contract:
+
+.. code-block:: python
+
+    # old_contract.vy
+    owner: public(address)
+    balanceOf: public(HashMap[address, uint256])
+
+.. code-block:: python
+
+    # new_contract.vy
+    owner: public(address)
+    minter: public(address)
+    balanceOf: public(HashMap[address, uint256])
+
+This would cause an issue when upgrading, as the `balanceOf` mapping would be located at `slot1` in the old contract, and `slot2` in the new contract.
```

Comment:
```suggestion
This would cause an issue when upgrading, as the ``balanceOf`` mapping would be located at ``slot1`` in the old contract, and ``slot2`` in the new contract.
```